internal/extension/extension.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed
+ * under the Apache License Version 2.0.
+ *
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2021 Datadog, Inc.
+ */
+
+package extension
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-lambda-go/internal/logger"
+)
+
+const (
+	// We don't want call to the Serverless Agent to block indefinitely for any reasons,
+	// so here's a configuration of the timeout when calling the Serverless Agent. We also
+	// want to let it having some time for its cold start so we should not set this too low.
+	timeout = 3000 * time.Millisecond
+
+	// helloUrl is the hello route url
+	helloUrl = "http://localhost:8124/lambda/hello"
+
+	// flushUrl is the flush route url
+	flushUrl = "http://localhost:8124/lambda/flush"
+)
+
+type ExtensionManager struct {
+	helloRoute         string
+	flushRoute         string
+	httpClient         HTTPClient
+	isExtensionRunning bool
+}
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func BuildExtensionManager() *ExtensionManager {
+	em := &ExtensionManager{
+		helloRoute: helloUrl,
+		flushRoute: flushUrl,
+		httpClient: &http.Client{Timeout: timeout},
+	}
+	em.checkAgentRunning()
+	return em
+}
+
+func (em *ExtensionManager) checkAgentRunning() {
+	req, _ := http.NewRequest(http.MethodGet, em.helloRoute, nil)
+	if response, err := em.httpClient.Do(req); err == nil && response.StatusCode == 200 {
```

Comment:
For the other runtimes, we opted to first do the file check for the binary, and then call the hello route. That was because they potentially had a timeout that extended the cold start duration

Reply:
✅